src/CacheManager.Core/Internal/BaseCache.cs

```diff
@@ -407,6 +410,55 @@ public virtual bool Remove(string key, string region)
         }
 
         /// <summary>
+        /// Gets all keys that match the given pattern.
+        /// </summary>
+        /// <param name="pattern">A glob to match against keys.</param>
+        /// <param name="region">The cache region.</param>
+        /// <returns>The matching unique keys in the cache</returns>
+        public virtual IEnumerable<string> Keys(string pattern, string region)
+        {
+            var keys = AllKeys();
```

Comment:
See above, this is only shared between the in-memory cases. The redis handler, for example, overrides Keys() and does not implement AllKeys()